core/src/main/kotlin/io/github/thanosfisherman/gasket/Gasket.kt

```diff
@@ -15,65 +16,39 @@ import kotlin.math.abs
 
 // Tolerance for calculating tangency and overlap
 private const val epsilon = 1f
-private val width = Gdx.graphics.width.toFloat()
-private val height = Gdx.graphics.height.toFloat()
 
-class Gasket(viewport: Viewport) {
+class Gasket(private var width: Float, private var height: Float) : Pool.Poolable {
     private var coneSegments = intArrayOf(4, 8, 20, 50).random()
     private var isConeShape = RandomXS128().nextInt(3) == 0
     private var colorRandomizer = ColorRandomizer()
-    private var circlesPool: Pool<Circle> = pool(1200) { Circle(viewport) }
+    private var circlesPool: Pool<Circle> = pool(1200) { Circle() }
     private var lastTimeCounted = TimeUtils.millis()
     private var sinceChange = 0f
 
     // Initialize first circle centered on canvas
-    private var c1 = circlesPool.obtain().also {
-        it.init(
-            width / 2,
-            height / 2,
-            -1 / (width / 2),
-            colorRandomizer.randomColor(),
-            coneSegments,
-            isConeShape
-        )
-    }
+    // Second circle positioned randomly within the first
+    // Third circle also positioned relative to the first
+    private lateinit var c1: Circle; private lateinit var c2: Circle; private lateinit var c3: Circle
 
-    private var r2 = randomFloatRange(100f, c1.radius / 2)
+    private var r2 = 0f; private var r3 = 0f
 
     // Generate a random angle between 0 and 2*PI
-    private var randomAngleRad: Float = randomFloatRange(0f, 2 * PI)
+    private var randomAngleRad: Float = 0f
 
     // Convert the angle to a unit vector
-    private var v = vec2(MathUtils.cos(randomAngleRad), MathUtils.sin(randomAngleRad))
-        .setLength(c1.radius - r2)
-
-    // Second circle positioned randomly within the first
-    private var c2 = circlesPool.obtain().also {
-        it.init(width / 2 + v.x, height / 2 + v.y, 1 / r2, colorRandomizer.randomColor(), coneSegments, isConeShape)
-    }
-
-    private var r3 = v.len()
-    private var v2 = Vector2(v).rotateRad(PI).setLength(c1.radius - r3)
-
-    // Third circle also positioned relative to the first
-    private var c3 = circlesPool.obtain().also {
-        it.init(width / 2 + v2.x, height / 2 + v2.y, 1 / r3, colorRandomizer.randomColor(), coneSegments, isConeShape)
-    }
+    private var v = Vector2(); var v2 = Vector2()
 
     // All circles in the gasket
-    private var allCircles = GdxArray<Circle>(false, 64).apply {
-        add(c1)
-        add(c2)
-        add(c3)
-    }
+    private var allCircles = GdxArray<Circle>(false, 64)
 
     // Queue for circles to process for next generation
-    private var queue = GdxArray<Triplet>(false, 64).apply {
-        // Initial triplet for generating next generation of circles
-        add(Triplet(c1, c2, c3))
+    private var queue = GdxArray<Triplet>(false, 64)
+
+    init {
+        reset()
     }
 
-    fun init() {
+    override fun reset() {
```

Comment:
So this basically saves us from having to type `gasket.init()` every time we obtain a new `Gasket()` right? Nice :)